typed/union.go

```diff
@@ -212,15 +215,15 @@ func newUnion(su *schema.Union) *union {
 
 // clear removes all the fields in map that are part of the union, but
 // the one we decided to keep.
-func (u *union) clear(m *value.Map, f field) {
+func (u *union) clear(m value.Map, f field) {
 	for _, fieldName := range u.f {
 		if field(fieldName) != f {
 			m.Delete(string(fieldName))
```

Comment:
Wait, what? If the maps can do this why is the field deleter so complex? This deserves a TODO at least?

Reply:
yeah, that part is not specifically that hard. It's just that we're not doing much in place right now.

Reply:
Needs a comment either way. It doesn't seem great for the deleter to implicitly do a copy + delete.